app/controllers/users_controller.rb

```diff
@@ -8,23 +8,49 @@ def new
     end
```

Comment:
for `def new`, use `redirect_to_role_portal` and make suggested changes to `redirect_to_role_portal` below